signer/core/signed_data.go

```diff
@@ -167,6 +168,23 @@ func (api *SignerAPI) determineSignatureFormat(ctx context.Context, contentType
 		// Clique uses V on the form 0 or 1
 		useEthereumV = false
 		req = &SignDataRequest{ContentType: mediaType, Rawdata: cliqueRlp, Messages: messages, Hash: sighash}
+	case apitypes.DataTyped.Mime:
+		// EIP-712 conformant typed data
+		var typedData apitypes.TypedData
+		if stringData, ok := data.(string); !ok {
+			return nil, useEthereumV, fmt.Errorf("input for %v must be an hex-encoded string", apitypes.DataTyped.Mime)
+		} else if jsonData, err := hexutil.Decode(stringData); err != nil {
+			return nil, useEthereumV, err
+		} else if err := json.Unmarshal(jsonData, &typedData); err != nil {
+			return nil, useEthereumV, err
+		}
+		if messages, err := typedData.Format(); err != nil {
+			return nil, useEthereumV, err
+		} else if sighash, rawData, err := apitypes.TypedDataAndHash(typedData); err != nil {
+			return nil, useEthereumV, err
```

Comment:
Looking at the return statements here, it seems like creating a new method

```
func (api *SignerAPI) typedDataRequest(data any) (*SignDataRequest, error)
```

would be better. Then you wouldn't need to have `useEthereumV` in every single error return.
BTW, the style here, using `else if`, is a bit weird.